### ISSUE TITLE
Executable size is big - idea(s) for reducing size.

### ISSUE THREAD

**Issue by junh1024** [d-fi/releases#42]
Hi.

The extracted exe is about 75mb, which is big.

I tried using UPX directly to compress the exe, but it [does not work](https://github.com/upx/upx/issues/2) for d-fi "Pkg: Error reading from file.", and the issues is unlikely to be fixed soon on the upx side.

Maybe you can compress the exe while compiling it with [pkgx](https://github.com/irelance/node-pkgx/) ? As a complete node & typescript rookie, I was unable to get it working in a few minutes. 

Maybe you can get pkgx working to compress the exe during the compile stage.

**Comment by nkeor** [comment]
From https://github.com/irelance/node-pkgx/:

> wine is need for linux or mac, since ResourceHacker only support pe

Not happening anytime soon...

The reason for such a big size is that the binary has to include a node runtime.

**Comment by junh1024** [comment]
Yes I know it needs a runtime. When you build, with pkg, do you compress it?

```
  – reduce size of the data packed inside the executable with GZip
    $ pkg --compress GZip index.js
```

[via](https://github.com/vercel/pkg#usage)

**Comment by nkeor** [comment]
Compression is not enabled right now, but unfortunately the binary includes a node runtime which isn't very compressable. The gains might be minimal. Anyways, I might enable it in a future release after some testing.

**Comment by sayem314** [comment]
[caxa](https://github.com/leafac/caxa) seems like a good candidate for packaging. It also supports `ESM` modules.

**Comment by nkeor** [comment]
Excelent, I'll look into it

**Comment by nkeor** [comment]
Well, caxa doesn't seem like such a good fit anymore:

```sh
$  yarn caxa -- --input . --output d-fi_caxa -- "{{caxa}}/node_modules/.bin/node" "{{caxa}}/dist/src/d-fi.js"
$ ls -lh d-fi_caxa
-rwxr-xr-x 1 namkhai namkhai 126M Apr  2 13:12 d-fi_caxa
$ ls -lh build/d-fi
-rwxr-xr-x 1 namkhai namkhai 46M Mar 30 11:59 build/d-fi
```

Other issues are

* needs /tmp/ mounted with `exec`
* fails to build with tmpfs' default inode count (runs out of inodes)

Using `--compression GZip` on pkg saves 3MB

**Comment by sayem314** [comment]
New version is built with Golang. Binary is is less than 8 MB.
